pkg/config/expand.go

```diff
@@ -535,10 +533,52 @@ func expandSimpleVariable(
 	// Verify output exists in module
 	found := slices.ContainsFunc(modInfo.Outputs, func(o modulereader.VarInfo) bool { return o.Name == ref.Name })
 	if !found {
-		return "", fmt.Errorf("%s: module %s did not have output %s",
+		return false, fmt.Errorf("%s: module %s did not have output %s",
 			errorMessages["noOutput"], refMod.ID, ref.Name)
 	}
-	return fmt.Sprintf("((module.%s.%s))", ref.ID, ref.Name), nil
+
+	return isInterGroupReference, nil
+}
+
+// Needs DeploymentGroups, variable string, current group,
+func expandSimpleVariable(
+	context varContext,
+	modToGrp map[string]int) (string, error) {
+
+	// Get variable contents
+	re := regexp.MustCompile(simpleVariableExp)
+	contents := re.FindStringSubmatch(context.varString)
+	if len(contents) != 2 { // Should always be (match, contents) here
+		err := fmt.Errorf("%s %s, failed to extract contents: %v",
+			errorMessages["invalidVar"], context.varString, contents)
+		return "", err
+	}
+
+	callingGroupID := context.blueprint.DeploymentGroups[context.groupIndex].Name
+	refStr := contents[1]
+
+	ref, err := identifySimpleVariable(refStr, callingGroupID)
+	if err != nil {
+		return "", err
+	}
+
+	isInterGroupReference, err := validateReference(ref, context, modToGrp)
```

Comment:
My preference for flow here (feel free to disregard if you disagree):

```go
// Checks based only on the string pattern. Could return a value or set a field of the struct
isInterGroupReference, err := ref.isInterGroupReference()

// This or include a field in ref for tracking intergroup or not
err = validateReference(ref, context, modToGrp, isInterGroupReference)
```

Reply:
My theory for `varReference` -- which does not have to be the only theory -- is that it's just converting the YAML string to a Go struct with well-defined fields. I am trying to avoid doing anything that is a "guess" (such as whether the target of the reference actually exists anywhere in the blueprint).

That said, because the source GroupID is a parameter to the function, we could add a field (or func interface) which would report whether the reference _superficially_ appears to be intergroup. I will look at the impact this would have on the code. It would probably not need the first line or the last parameter in the 2nd. I'd still want to be validating the intergroup-ness of a reference.

Reply:
I have modified varReference.Explicit to be varReference.ExplicitInterGroup. This value is validated by `validateReference` and can then be trusted by remaining code.